app_build_suite/build_steps/giant_swarm_validators/helm.py

```diff
@@ -0,0 +1,86 @@
+"""
+General idea for naming validation checks:
+- comply to 1 letter + 4 digits pattern
+- check types and first letters:
+  - file system layout and structure: "F"
+  - Chart.yaml related problems "C"
+"""
+import argparse
+import logging
+import os
+import re
+
+import yaml
+from step_exec_lib.errors import Error
+
+from app_build_suite.build_steps.helm_consts import VALUES_SCHEMA_JSON, CHART_YAML, TEMPLATES_DIR, HELPERS_YAML
+
+logger = logging.getLogger(__name__)
+
+
+ANNOTATIONS_KEY = "annotations"
+GS_TEAM_LABEL_KEY = "application.giantswarm.io/team"
+
+
+class GiantSwarmValidatorError(Error):
+    pass
+
+
+class HasValuesSchema:
+    def get_check_code(self) -> str:
+        return "F0001"
+
+    def validate(self, config: argparse.Namespace) -> bool:
+        return os.path.exists(os.path.join(config.chart_dir, VALUES_SCHEMA_JSON))
+
+
+class HasTeamLabel:
+
+    escaped_label = re.escape(GS_TEAM_LABEL_KEY)
+    _label_regexp = (
```

Comment:
Do we need to have a regex because the file isn't proper YAML or why can't we just parse the file normally?